### PR TITLE
Refuse sending card if process or state does not exist (#3234)

### PR DESCRIPTION
**Pull request by vlo-rte**
- In release note :
  -  In chapter :  Features
  -  Text : #3234 : Refuse sending card if process or state does not exist

### CODE REVIEW

File: services/cards-publication/src/main/java/org/opfab/cards/publication/controllers/CardController.java [hunk 1]
```diff
@@ -56,6 +60,17 @@ public class CardController {
...
+
+
+        if ((user != null) && (!cardProcessingService.doesProcessStateExistInBundles(card.getProcess(),
+                card.getProcessVersion(), card.getState(), user.getUserData().getLogin()))) {
+            throw new ApiErrorException(
+                    ApiError.builder()
+                            .status(HttpStatus.BAD_REQUEST)
+                            .message(String.format(UNEXISTING_PROCESS_STATE, card.getProcess(), card.getState(), card.getProcessVersion(), card.getProcessInstanceId()))
+                            .build());
+        }
```
Comment (freddidierRTE) [inline services/cards-publication/src/main/java/org/opfab/cards/publication/controllers/CardController.java:72]: Should be done in cardProcessingService like other controls

File: src/test/api/karate/businessconfig/resources/packageBundles.sh [hunk 2]
```diff
@@ -13,4 +13,12 @@ cd ..
 cd bundle_api_test_apogee 
 tar -czvf bundle_api_test_apogee.tar.gz config.json i18n.json css/ template/ 
 mv bundle_api_test_apogee.tar.gz ../
+cd ..
+cd bundle_defaultProcess_V1
+tar -czvf bundle_defaultProcess_V1.tar.gz config.json i18n.json css/ template/
+mv bundle_defaultProcess_V1.tar.gz ../
+cd ..
```
Comment (freddidierRTE) [inline src/test/api/karate/businessconfig/resources/packageBundles.sh:20]: Must we really copy all files , could we just take the file form the resource directory ?

File: client/businessconfig/src/main/java/org/opfab/businessconfig/model/AcknowledgmentAllowedEnum.java [hunk 3]
```diff
@@ -32,4 +34,20 @@ public enum AcknowledgmentAllowedEnum {
...
+    }
+
+    @JsonCreator
+    public static AcknowledgmentAllowedEnum fromValue(String text) {
+        for (AcknowledgmentAllowedEnum b : AcknowledgmentAllowedEnum.values()) {
+            if (String.valueOf(b.value).equals(text)) {
+                return b;
+            }
+        }
+        return null;
```
Comment (freddidierRTE) [inline client/businessconfig/src/main/java/org/opfab/businessconfig/model/AcknowledgmentAllowedEnum.java:51]: Why this is needed ?

**Review by freddidierRTE — CHANGES_REQUESTED** [review]

File: client/businessconfig/src/main/java/org/opfab/businessconfig/model/AcknowledgmentAllowedEnum.java (hunk 3 above)
Comment (vlo-rte) [reply client/businessconfig/src/main/java/org/opfab/businessconfig/model/AcknowledgmentAllowedEnum.java]: I can't figure out why I needed it. Weird...
(So I remove)

File: src/test/api/karate/businessconfig/resources/packageBundles.sh (hunk 2 above)
Comment (vlo-rte) [reply src/test/api/karate/businessconfig/resources/packageBundles.sh]: Done.

**Pull request merged**